File: app/src/main/java/com/example/savioguedes/testeandroid/login/LoginContract.java
package com.example.savioguedes.testeandroid.login;

import com.example.savioguedes.testeandroid.model.Login;

public interface LoginContract {

    interface View{

        void initView();
        void passUserinfo(int id, String name, String account, String balance);
        void showProgressDialog();
        void onErroRequest();
    }

    interface Presenter{

        void getLoginExecute(Login login);
        void setPreferences(String value, String tag);
        boolean isValidFields(String user, String password);
        String getPreferences(String tag);
    }
}

File: app/src/main/java/com/example/appbanksantander_accenturetalentacquisition/Presenter/Login/LoginContract.kt
package com.example.appbanksantander_accenturetalentacquisition.Presenter.Login

import com.example.appbanksantander_accenturetalentacquisition.Model.UserAccountModel

interface LoginContract {
    interface View{
        fun UserDetails(userAccountModel: UserAccountModel)
    }
    interface UserActionListener{
        fun loadUser()
        fun saveUser(user: String)
        fun verifyUser(): String
    }
}